### ISSUE TITLE
zip icon abnormally [BUG] [GUI]

### ISSUE THREAD

**Issue by xzhouzhou758** [aonez/Keka#1177]
- Keka version: [ 1.2.57 (5068)]
- macOS version: [macOS 13.0 (22A380)]

The icon is displayed normally in the download folder, but the icon is displayed abnormally in the download of the dock

<img width="199" alt="截屏2022-11-05 18 41 34" src="https://user-images.githubusercontent.com/52977053/200116093-ab9a3c48-dd8b-409e-88d1-1826cc29c5de.png">
<img width="142" alt="截屏2022-11-05 18 41 23" src="https://user-images.githubusercontent.com/52977053/200116097-cf2be7fe-3864-4d9f-bd72-f84a1d373eae.png">

**Comment by desmortum** [comment]
Yeah. Same thing. In Finder looks good. In Dock looks generic in folders like Downloads or if you pin file to Dock. Also, looks generic in Chrome downloads. Seems like something changed in Ventura.

<img width="858" alt="Screenshot 2022-11-05 at 17 58 12" src="https://user-images.githubusercontent.com/38209369/200132112-57ed93dd-bf66-4053-b132-94ba87ca9391.png">
<img width="306" alt="Screenshot 2022-11-05 at 17 58 26" src="https://user-images.githubusercontent.com/38209369/200132115-eb4c54da-5ae0-4517-8de2-dd09ea9b435a.png">
<img width="621" alt="Screenshot 2022-11-05 at 17 58 01" src="https://user-images.githubusercontent.com/38209369/200132117-6f0d4cc0-1dba-4cd5-b0a9-6229dea8db52.png">

**Comment by desmortum** [comment]
Same for *.tar. I guess all formats are affected. I tried cleaning icon cache and doing clean reinstall of Keka. When you remove Keka, original archiver icons are back in Dock and Chrome. When you install Keka, they're generic again.

**Comment by aonez** [comment]
Seems it’s #453 again. Will check if I can replicate this.

**Comment by desmortum** [comment]
Unlike in #453, icon is completely blank. As if all apps except Finder cannot get icon associated with Keka formats at all.

**Comment by desmortum** [comment]
I've done some experimentation and it seems no privacy permission changes help. What I did find is that other built-in and third-party apps have their associated type icons shown properly. For example Visual Studio *.sln file works fine.

<img width="283" alt="Screenshot 2022-11-05 at 20 06 05" src="https://user-images.githubusercontent.com/38209369/200136824-398f7143-9605-418b-ad26-89236177f56d.png">

**Comment by desmortum** [comment]
Found few [discussions](https://discussions.apple.com/thread/254326530) on Apple site. It seems Adobe PDF is affected too. Hope you can find a way to fix it.

**Comment by desmortum** [comment]
@aonez, any luck replicating?

**Comment by aonez** [comment]
Tried with one of the first macOS Ventura betas. While it does not show a generic white document icon, the Dock representations show the macOS zip white icons even if Keka is the default handler. As you can see in the screenshots, same as in #453, only compressed files already recognized by macOS have this issue. Brotli, LZIP and ZSTD do not present this issue.

Will install and try with the current Ventura release, but most probably this should be fixed in Apple's end. I will report it myself after trying with the current version.

This should be reported using the Feedback Assistant: https://developer.apple.com/bug-reporting

![Screenshot 2022-11-09 at 17 13 41](https://user-images.githubusercontent.com/480163/200882375-3598675f-6b91-4507-b82a-c9cb57626046.png)

![Screenshot 2022-11-09 at 17 13 15](https://user-images.githubusercontent.com/480163/200882352-de17b92e-1fd2-40c9-8d43-78c240a8f989.png)

**Comment by desmortum** [comment]
Yeah, Brotli is okay in release version of Ventura.

<img width="525" alt="Screenshot 2022-11-09 at 17 51 23" src="https://user-images.githubusercontent.com/38209369/200891536-ad412d87-7b6e-4325-a3b2-2d4cb5fb9768.png">

**Comment by aonez** [comment]
So tried with macOS 13.1 beta and it shows the white icons as you reported. Even when extracting with Keka they show white, same as in Chrome.

This is the [offending API](https://developer.apple.com/documentation/appkit/nsworkspace/1528158-iconforfile):
```
[NSWorkspace.sharedWorkspace iconForFiles:(nonnull NSArray<NSString *> *)];
```

I'm reporting this to Apple right away, but if #453 meant something, this might be not fixed until macOS 14 or later. Fingers crossed they fix this before that.

![Screenshot 2022-11-10 at 10 30 28](https://user-images.githubusercontent.com/480163/201054509-b1f7e237-d70f-4661-bed8-14cf6307950b.p
![Screenshot 2022-11-10 at 10 40 27](https://user-images.githubusercontent.com/480163/201056941-ea029e04-0b84-4105-b1ca-c7b506f1e941.png)
ng)

**Comment by aonez** [comment]
Also note this will probably affect all apps that override macOS defined icons, such as video apps (VLC or INA).

**Comment by aonez** [comment]
Already reported to Apple:

```
Title: Third party document icons not displayed properly on macOS 13
Area: AppKit
Issue: Incorrect/Unexpected behaviour

Third party applications that define icons for macOS recognized document types are not displayed properly on macOS 13.

To reproduce:

- Download Keka or another app that defines icons for system known formats (like ZIP, MOV)
- Be sure to set that application as the default handler for the document type (ZIP, MOV).
- Restart the system if needed to refresh icon caches.
- The custom icons will show properly in the Finder but will show blank where they’re displayed using [NSWorkspace.sharedWorkspace iconForFile:(NSString *)].

You can check this adding a folder with the offending files inside to the Dock, as shown in the screenshots.

Additional info:

This was reported by three users already (more to come) and reproduced by me.
https://github.com/aonez/Keka/issues/1177

All apps that use the NSWorkspace iconForFile: API (and variations of it) will be affected by this issue.
https://developer.apple.com/documentation/appkit/nsworkspace/1528158-iconforfile

macOS 13.0 Beta 22A5286j displayed the macOS icons, instead of the custom ones.
macOS 13.0 22A380 and later (tested macOS 13.1 Beta 22C5044e as well) display blank icons.
```

Please report it too if you are affected by this issue and want it to be fixed as soon as possible.

**Comment by desmortum** [comment]
Reported as well.

**Comment by welaxxx** [comment]
I've been trying to fix this problem from day 1 in venture with no luck but I found something maybe it helps...

I try many App instead of Keka to see if the problem from Keka or Mac Os,more than 100 App, same problem same issue 
except 1 App, called "**Oka Unarchiver PRO**".  https://apps.apple.com/am/app/oka-unarchiver-unzip-zip-rar/id1441507725?mt=12   - free and paid 

everything works fine with this App and I don't know why????!!!!

maybe this helps who knows

**Comment by desmortum** [comment]
@aonez, 13.1 beta 3 is out. Maybe if you have time and partition to check it out give it a shot.

**Comment by aonez** [comment]
There's no mention to this in the [release notes](https://developer.apple.com/documentation/macos-release-notes/macos-13_1-release-notes), but I'll check it anyway.

**Comment by aonez** [comment]
Same same...

![Screenshot 2022-11-16 at 21 15 23](https://user-images.githubusercontent.com/480163/202284969-a55b7959-1456-42ac-beed-21a7f51f272c.png)

**Comment by desmortum** [comment]
Thanks for checking.

**Comment by aonez** [comment]
Someone that also found the issue:

https://twitter.com/mjtsai/status/1598433805777379336?s=46&t=Mxh_95-9ojAa8c1J2tB4_w

**Comment by pan93412** [comment]
Reported as well! Hope Apple can take care of their products' issues. I have reported for at least 3 issues related to macOS 13 😢 

<img width="739" alt="截圖 2022-12-02 上午9 47 14" src="https://user-images.githubusercontent.com/28441561/205196151-9cb372d3-ea54-4d1b-b045-999ed6805567.png">

**Comment by aonez** [comment]
Ventura 13.2 Beta still has the issue, as expected.

![image](https://user-images.githubusercontent.com/480163/208073127-5ac3f9e3-85f7-477a-b13d-340ed0630c95.png)

**Comment by pan93412** [comment]
Ventura 13.2 Release still has the issue :(

<img width="151" alt="image" src="https://user-images.githubusercontent.com/28441561/215112406-c0471d60-2141-40da-a3e8-9fd757d97677.png">

**Comment by melyux** [comment]
Reported too. But does Apple even take action on any feedback items? It's like a black hole. I've probably reported 20+ things in the past but they were never fixed unless Apple decided to revamp the entire application and fixed it by chance several years later.

**Comment by codebysandwich** [comment]
Ventura 13.2.1 Release still has the issue :(, alfred affected too.

**Comment by despecial** [comment]
Ventura 13.3 still does not fix this :(

**Comment by JustCrazyer** [comment]
Ventura 13.3.1 still does not fix this :(

**Comment by aonez** [comment]
As reported by @pan93412 we got some news in the Feedback Assistant that reads:

```
Recent Similar Reports: Less than 10
Resolution: Potential fix identified - For a future OS update
```

So thanks to @pan93412 for the this and to all who reported this. Expect the fix in macOS 14, as expected 🙃

**Comment by animusastralis** [comment]
Finally! Ventura 13.4 got it fixed

**Comment by aonez** [comment]
I'm still seeing the issue on Ventura 13.4 (22F66)

<img width="207" alt="Screenshot 2023-05-19 at 13 07 26" src="https://github.com/aonez/Keka/assets/480163/a0736763-20e6-4f8c-a4b3-1c2672a4abb9">

**Comment by desmortum** [comment]
For me it's also fixed.
<img width="734" alt="Screenshot 2023-05-19 at 13 09 51" src="https://github.com/aonez/Keka/assets/38209369/a04f7e85-017a-4241-a4ce-cc702283f6e4
<img width="246" alt="Screenshot 2023-05-19 at 13 10 08"
<img width="223" alt="Screenshot 2023-05-19 at 13 09 43" src="https://github.com/aonez/Keka/assets/38209369/0d97307d-da2f-4993-a797-373789a13405">
 src="https://github.com/aonez/Keka/assets/38209369/117cf592-7f44-4cfb-9c43-86debef5b20e">
">

<img width="370" alt="Screenshot 2023-05-19 at 13 09 57" src="https://github.com/aonez/Keka/assets/38209369/bc17760c-0701-4a03-b175-b80dfb613b93">

**Comment by desmortum** [comment]
@aonez, try rebooting or doing "killall Dock"

**Comment by aonez** [comment]
@desmortum yep already did a restart, cold start and relaunched Finder... But also in Safari the bug persists. @animusastralis are you on an Intel too?
<img width="334" alt="Screenshot 2023-05-19 at 13 14 00" src="https://github.com/aonez/Keka/assets/480163/4126df20-f8d1-4da3-ba19-b85bf230c35b">

**Comment by desmortum** [comment]
@aonez, maybe OS icon cache?

**Comment by aonez** [comment]
Yeah probably, just updated an ARM VM and it's indeed fixed 👏🏼

**Comment by animusastralis** [comment]
@aonez no, I'm on M2 MacBook Pro. I didn't noticed it right away and did a couple of reboots for other reasons

**Comment by pan93412** [comment]
Before clearing cache, it did still not work:

<img width="554" alt="image" src="https://github.com/aonez/Keka/assets/28441561/af6c33b7-6a25-4894-ae59-865463e8c940">

After [clearing the cache with a Gist script](https://gist.github.com/ismyrnow/e92c6010cda9325b2d8811387a05f224), it works now:

<img width="239" alt="image" src="https://github.com/aonez/Keka/assets/28441561/53688966-279a-40a2-8321-57cd894099bb">

@aonez

**Comment by gingerbeardman** [comment]
Congrats everybody! what a long journey.

**Comment by WHTJEON** [comment]
This doesn't fix blank icons in Transmit5. Does anyone know a fix for Transmit?
